src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -281,4 +297,11 @@ public abstract partial class RegexRunnerFactory
         protected RegexRunnerFactory() { }
         protected internal abstract System.Text.RegularExpressions.RegexRunner CreateInstance();
     }
+    public ref struct SplitEnumerator
+    {
+        public ReadOnlySpan<char> Current { get; private set; }
+        public SplitEnumerator GetEnumerator() => this;
```

Comment:
I'm not aware of any other cases in the BCL where we have a GetEnumerator method like this on a struct enumerator.  I understand you want it to be able to directly foreach the results without introducing an enumerable struct to serve as the return type, but I'm not sure this is a pattern we want to introduce.  You should be sure to highlight this as part of any API review discussion.

Reply:
Right. Thanks, I'll put it into the initial post as to discuss.